plugins/postgresql.yaml

```diff
@@ -36,26 +36,11 @@ pipeline:
   - id: general_regex_parser
     type: regex_parser
     parse_from: log_entry
-    regex: '^t=(?P<time>\d{4}-\d{2}-\d{2}\s+\d{2}:\d{2}:\d{2}\s+[^\s]+)\s+p=(?P<process_id>\d*)\s+s=(?P<process_start_time>[^\.]*)\.[^\s]+\s+l=(?P<process_log_line>[^\s]*)\s+u=(?P<user_name>[^\s]*)\s+db=(?P<database>[^\s]*)\s+r=(?P<client_address>[^\s]*)\s*(LOG:\s*duration:\s*(?P<duration>[\w\.:]*)\s*ms\s*)?(?P<message>[\w\W]+)?'
+    regex: '^(?P<time>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{3,} \w+)\s*\[(?P<tid>\d+)\](?:\s+(?P<role>\S*)@(?P<user>\S*))?\s*(?P<level>\w+):\s+(?P<message>.*)'
     timestamp:
       parse_from: time
       layout: '%F %T %Z'
 
-  - type: router
-    default: client_address_regex_parser
-    routes:
-      - output: client_address_regex_parser
-        expr: '$record.duration != ""'
-        labels:
-          log_type: 'postgresql.slow_query'
-
-  - id: client_address_regex_parser
-    if: '$record.client_address matches "[\\w\\.]*\\([\\d]*\\)"'
-    type: regex_parser
-    parse_from: $record.client_address
-    regex: '(?P<client_address>[^\(]*)\((?P<client_address_port>[^\)]*)\)'
-    output: statement_regex_parser
-
```

Comment:
Is there going to be a gap here?

Reply:
I couldn't figure out any logs that actually have that. I guess I could make it an optional capture group for it?

Reply:
I think the slow query log would look like this
```
2018-08-20 08:19:28.151 CEST [22845] LOG: duration: 10010.353 ms statement: SELECT pg_sleep(10);
```

https://www.cybertec-postgresql.com/en/3-ways-to-detect-slow-queries-in-postgresql/

Reply:
Updated 
tested with 
```
2022-02-17 21:07:07.829 UTC [5529] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2022-02-17 21:07:07.835 UTC [5530] LOG:  database system was shut down at 2022-02-17 21:07:07 UTC
2022-02-17 21:07:07.841 UTC [5529] LOG:  database system is ready to accept connections
2018-08-20 08:19:28.151 CEST [22845] LOG: duration: 10010.353 ms statement: SELECT pg_sleep(10);
```

output
```
2022-02-17 21:07:07.829 UTC [5529] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2022-02-17 21:07:07.835 UTC [5530] LOG:  database system was shut down at 2022-02-17 21:07:07 UTC
2022-02-17 21:07:07.841 UTC [5529] LOG:  database system is ready to accept connections
2018-08-20 08:19:28.151 CEST [22845] LOG: duration: 10010.353 ms statement: SELECT pg_sleep(10);
```

Reply:
@jsirianni ^